src/services/dashboard.js

```diff
@@ -7,11 +7,14 @@ import { getAllMonthsWithinRange } from '../utils/moment_constants';
 const { endpoints } = window;
 
 function scrollUntilEmpty(data) {
-  const endpoint = `${endpoints.elasticsearch}/_search/scroll?scroll=1m`;
+  const endpoint = `${endpoints.pbench_server}/elasticsearch`;
   const allData = data;
+  const indices = `_search/scroll?scroll=1m&scroll_id=${allData._scroll_id}`;
```

Comment:
Usually, the first `scroll` API call established a context, returning a `scroll_id`, which is used on subsequent calls.  Is that still happening?

Reply:
Not sure what you meant by that, this is just a structural change separating an es endpoint from indices without any logical changes. In the original code we are querying like this
```
const endpoint = `${endpoints.elasticsearch}/_search/scroll?scroll=1m`;
const scroll = request.post(`${endpoint}&scroll_id=${allData._scroll_id}`);
```
However with server api we need a `data` field with a post payload so we do the following specifying the indices separately 
```
endpoint = `${endpoints.pbench_server}/elasticsearch`;
const indices = `_search/scroll?scroll=1m&scroll_id=${allData._scroll_id}`;
const scroll = request.post(endpoint, {
      data: { indices },
    });

Reply:
Isn't there an `if` on line 13 where the `scroll_id` is only added if it exists in the returned data?

The `scroll_id` seems to only be added at line 19 below.

Reply:
I think that Javascript will just return `undefined` at line 12 if the `_scroll_id` property is missing, and in that case the `if` at line 14 won't trigger, so the `&scroll_id=undefined` won't be used.

But perhaps Gurbir or Aquib can confirm the Javascript behavior more confidently...

Reply:
Actually, I just pulled up Chrome's debug console and tried it; the undefined JSON field does simply resolve as "undefined":

```
x = {a: "test", b: "foo"}
y = `?scroll_id=${x.bar}`

"?scroll_id=undefined"
```

So I'm pretty sure this is fine.

Reply:
What does the Elasticsearch V7 scroll API do when it receives `undefined` as the `scroll_id`?

Aside from the fact that specifying the `scroll_id` in the parameters of the URL is deprecated as of 7.0, has anybody tested this change to make sure it behaves as expected?

See https://www.elastic.co/guide/en/elasticsearch/reference/current/scroll-api.html.

Reply:
Okay I fixed it by creating indices inside if block so `scroll_id` should never be undefined in the `indices`

Reply:
Okay, it appears that `scrollUntilEmpty` is only ever called after an initial `_search/scroll` API is made, so this appears to be correct.

Though, it would be nice if somebody can confirm this actually still works.

Reply:
I just did some experimenting with direct Elasticsearch queries, and id `15ed6b30e0f92df4fffdba7bae72c41d` (`fio_rw_2021.03.03T02.04.39`) on the staging server is returning a `_scroll_id` for `queryIterationSamples`, so the code ought to be triggering if Nikhil runs his dashboard code and drills down through that dataset.